osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/GetPeriodicMeterReadsCommandExecutorTest.java

```diff
@@ -0,0 +1,111 @@
+/**
+ * Copyright 2019 Smart Society Services B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.opensmartgridplatform.adapter.protocol.dlms.domain.commands;
+
+import static java.util.Arrays.asList;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.same;
+import static org.mockito.Mockito.when;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.GregorianCalendar;
+
+import org.assertj.core.api.Assertions;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.openmuc.jdlms.AttributeAddress;
+import org.openmuc.jdlms.GetResult;
+import org.openmuc.jdlms.ObisCode;
+import org.openmuc.jdlms.datatypes.DataObject;
+import org.opensmartgridplatform.adapter.protocol.dlms.domain.entities.DlmsDevice;
+import org.opensmartgridplatform.adapter.protocol.dlms.domain.factories.DlmsConnectionHolder;
+import org.opensmartgridplatform.adapter.protocol.dlms.infra.messaging.DlmsMessageListener;
+import org.opensmartgridplatform.dto.valueobjects.smartmetering.PeriodTypeDto;
+import org.opensmartgridplatform.dto.valueobjects.smartmetering.PeriodicMeterReadsRequestDto;
+
+@RunWith(MockitoJUnitRunner.class)
+public class GetPeriodicMeterReadsCommandExecutorTest {
+    private static final AttributeAddress CLOCK = new AttributeAddress(8, new ObisCode(0, 0, 1, 0, 0, 255), 2);
+
+    private GetPeriodicMeterReadsCommandExecutor executor;
+
+    @Mock
+    private DlmsMessageListener listener;
+
+    @Mock
+    private DlmsHelperService helperService;
+
+    @Mock
+    private AttributeAddressHelperService attributeAddressHelperService;
+
+    @Mock
+    private AmrProfileStatusCodeHelperService amrProfileStatusCodeHelperService;
+
+    private DlmsConnectionHolder connectionHolder;
```

Comment:
It seems this class was renamed previously from `DlmsConnectionManager`. I think the current name is very confusing since a 'Holder' is typically a class which uses some form of the Initialization-on-demand holder idiom, usually involving a singleton instance and/or static access. Just noticed it, probably not something you should fix as part of this task.

Reply:
It is renamed to DlmsConnectionManager. This branch is now also updated to this name.